safe_control_gym/controllers/mpc/mpc_acados.py

```diff
@@ -260,38 +260,120 @@ def processing_acados_constraints_expression(self,
         ocp.constraints.ubu = self.env.constraints.input_constraints[0].upper_bounds
         ocp.constraints.idxbu = idxbu # active constraints dimension
         '''
-
-        ub = {'h': set_acados_constraint_bound(h_expr, 'ub', self.constraint_tol),
```

Comment:
I briefly compared these two implementations, and I find yours is faster. 
Updated version:
![image](https://github.com/user-attachments/assets/45ab7432-4a97-4695-9e33-689562a642ce)
Before:
![image](https://github.com/user-attachments/assets/059afcec-9082-4e1d-97ab-4860a4740987)

I'd like to learn more about what case 1 2 3 means.  Also I didn't fully understand what `detect_constraint_structure` does. It seems that it doesn't return anything (or could you ask your colleague to add some docstring for this function.)

Apart from that, I wonder whether we can condense these codes more. For example I think gym has a fixed way to parse the constraint and we might always have lower bounds before upper bounds ([here](https://github.com/utiasDSL/safe-control-gym/blob/5f20bac6d642a00c81fca9823bcd0b05fa118275/safe_control_gym/envs/constraints.py#L321C1-L322C63)). So we might could safely remove the flag `start_with_ub`. Please correct me if I am wrong @adamhall .

Reply:
I think for IPOPT the benefit would the very minimal for the computation speed on small problems. But for large problems if the constraints are double-sided and you implement them as `h(x, u) < 0` you basically double the number of constraints. So it would be nice indeed to preserve the constraints structure.